app/models/mixins/purging_mixin.rb

```diff
@@ -98,8 +98,36 @@ def purge_by_scope(older_than = nil, window = nil, &block)
       _log.info("Purging #{table_name.humanize}...Complete - Deleted #{total} records")
     end
 
+    def purge_by_orphaned(fk_name, window = purge_window_size)
+      _log.info("Purging orphans in #{table_name.humanize}...")
+      total = purge_orphans(fk_name, window)
+      _log.info("Purging orphans in #{table_name.humanize}...Complete - Deleted #{total} records")
+      total
+    end
+
     private
 
+    def purge_orphans(fk_name, window)
+      # For now, this only supports polymorphic references
+      # We don't currently have a situation where a table with a regular reference needs to be purged
+      polymorphic_type_column = "#{fk_name}_type"
+      total = 0
+
+      polymorphic_classes(polymorphic_type_column).each do |klass|
+        resource_table = connection.quote_table_name(klass.table_name)
+
+        scope = joins("LEFT OUTER JOIN #{resource_table} ON #{table_name}.resource_id = #{resource_table}.id")
+                .where(resource_table => {:id => nil})
+                .where("#{table_name}.#{connection.quote_column_name(polymorphic_type_column)} = #{connection.quote(klass.name)}")
```

Comment:
can you just do?
```ruby
where(polymorphic_type_column => klass.name)
```

Reply:
Brakeman complained about that :worried: